mypy/checkexpr.py

```diff
@@ -5980,6 +5990,22 @@ def accept(
                 typ = self.visit_conditional_expr(node, allow_none_return=True)
             elif allow_none_return and isinstance(node, AwaitExpr):
                 typ = self.visit_await_expr(node, allow_none_return=True)
+            # Deeply nested generic calls can deteriorate performance dramatically.
+            # Although in most cases caching makes little difference, in worst case
+            # it avoids exponential complexity.
+            # TODO: figure out why caching within lambdas is fragile.
```

Comment:
I also discovered that in #19408. Accepting a lambda has explicit type context dependency (`infer_lambda_type_using_context`), so generic calls do really require re-accepting the lambda every time, context from outer generic may have propagated later.

Reply:
OK, thanks for the pointer. I will update the comment.